### PR TITLE
[Review] Request from 'schubi2' @ 'yast/skelcd-control-CASP/review_161207_fate_321764'

### PR DESCRIPTION
**Pull request by schubi2**
Patching of /etc/zypp/zypp.conf will be done while generating the installation image now.

### CODE REVIEW

**Comment by jreidinger** [comment]
to be honest I do not like much this approach as:

1. no test
2. easy to break in future
3. shell, which make future extension harder

why not having e.g. common client that do such configuration, can be tested and then simply place such client into installation workflow?

File: control/control.CASP.xml [hunk 1]
```diff
@@ -463,6 +468,17 @@ defined by a role can be overridden in the next steps if necessary.&lt;/p&gt;</l
                 <module>
                     <label>Load Linuxrc Network Configuration</label>
                     <name>install_inf</name>
+                    <!-- Configuring libzypp in order to install a minimal system (no recommended packages,...) Fate#321764 -->
```
Comment (imobachgs) [inline control/control.CASP.xml]: Is the control file the correct place to set this? Maybe YaST is the proper place. I'm not saying that it's wrong but I expected this to be anywhere else :)

**Comment by schubi2** [comment]
Well, that would another possibility too. I have had also in mind. The advantage of that solution would be:

- located in one file only. You do not need additional clients which should be in an own package because it is
CASP only. I think it is not a good idea to put it into e.g. yast2-installation.
- For me it is a little bit over engineered to put these 3 calls in an own infrastructure. I have thought that the
  script sections in control.xml file is intent to use for such kind of changes.
- We would need two clients. One for patching the zypp.conf in the inst_sys and one (e.g. zypp_patch_finish.rb) to patch the zypp.conf in the installed system too. So there are a lot of places where we have to touch the code.

But I am open for your suggestions. Where should this client be located ?

**Comment by schubi2** [comment]
The entries of a patched zypp.conf are:

# minimal CASP; set by YaST control.xml
rpm.install.excludedocs = yes

The the admin/devel. knows where these entries come from

**Comment by jreidinger** [comment]
@schubi2 I think it is fine to have it in yast2-installation, as we need to place there also other CASP dialogs, that we need to use. Even now in installation are clients that are used only in very specific case.
I think that such script section main intention is to allow add-ons to do some dirty dark magic, which should not be case for product.

Regarding two clients, if you use same approach, you just need to patch insts-sys one and then add line to https://github.com/yast/skelcd-control-CASP/blob/cb96171a5778bb65e852f13797646097108272b8/control/control.CASP.xml#L49 that store insts-sys zypp.conf

BTW regarding modified zypp.conf. I still more like to have it ( as already mentioned during planning ) in skelcd-casp as when you start yast, there is always risk, that it will be already initialized and changes ignored.

**Comment by jreidinger** [comment]
ah, not skelcd, but I mean installation-images - https://build.suse.de/package/show/SUSE:SLE-12-SP2:Update:Products:CASP10/installation-images-SLES

File: control/control.CASP.xml [hunk 2]
```diff
@@ -122,6 +122,11 @@ textdomain="control"
 
         <!-- BNC #435479: Show support status in package details -->
         <display_support_status config:type="boolean">true</display_support_status>
+
+        <!-- AutoYaST clone will be called in the installed system. Unfortunately there is no information about handling recommended
+             packages while first installation. This setting will be taken for install_recommended in the software section
+             of AY configuration file. Default is "true" -->
```
Comment (imobachgs) [inline control/control.CASP.xml]: What about adding the fate number here?

**Review by imobachgs — APPROVED** [review]

File: control/control.CASP.xml (hunk 2 above)
Comment (imobachgs) [reply control/control.CASP.xml]: Well, it's not needed at all as we're using the number for the simplified workflow.

**Review by imobachgs — APPROVED** [review]

**Pull request merged**